test/integration/client_test.ts

```diff
@@ -245,4 +250,408 @@ describe('Client', function () {
     unsub1();
     unsub2();
   });
+
+  it('client event flow', async function () {
```

Comment:
This test is complex, so it would be nice to write the description. Is this test related to the figure in the issue?

Reply:
Yes. It tests whether `client.subscribe` correctly detects the events when the client performs `activate`, `attach`, `updatePresence`, `detach`, and `deactivate`. The testing proceeds according to the flowchart illustrated in the figure.